sdk/state/state.go

```diff
@@ -36,17 +36,16 @@ type Channel struct {
 	// TODO - leave execution out for now
 	// iterationNumberExecuted int64
 
-	// The balance owing from the initiator to the responder, if positive, or
-	// the balance owing from the responder to the initiator, if negative.
-	// TODO - use Balance struct
-	amount Amount
-
 	initiator           bool
 	localEscrowAccount  *EscrowAccount
 	remoteEscrowAccount *EscrowAccount
 
 	localSigner  *keypair.Full
 	remoteSigner *keypair.FromAddress
+
+	lastConfirmedPayment *Payment
+	// TODO - need to store this too?
+	// lastUnConfirmedPayment *Payment
```

Comment:
@leighmcculloch why again do we need to store lastUnconfirmed? Is this something that will be useful later?

Reply:
Once a participant has signed C or D for an iteration number they really shouldn't sign a new payment with the same iteration number. So we need to store something, storing the incomplete payment seems possibly useful and really easy to understand as a reader of the code. We might be able to optimize it and store less but we don't need to optimize that yet.

Reply:
I've opened an issue for this specifically: https://github.com/stellar/experimental-payment-channels/issues/69.

Reply:
ok thx, I'll leave commented for this PR then